src/extra/gpkg.jl

```diff
@@ -0,0 +1,192 @@
+    throw(ErrorException("missing required metadata tables in the GeoPackage SQL database"))
+  end
+
+  # According to https://www.geopackage.org/spec/#r13
+  # A GeoPackage SHALL include a 'gpkg_contents' table
+  if isnothing(SQLite.tableinfo(db, "gpkg_contents"))
+    throw(ErrorException("missing required metadata tables in the GeoPackage SQL database"))
+  end
+
+  db
+end
+
+function gpkgextract(db; layer=1)
+  # get the feature table given the layer number
+  layerinfo = DBInterface.execute(
+    db,
+    """
+    SELECT g.table_name AS tablename, g.column_name AS geomcolumn, g.z,
+    c.srs_id AS srsid, srs.organization AS org, srs.organization_coordsys_id AS code
+    FROM gpkg_geometry_columns g, gpkg_spatial_ref_sys srs
+    """ *
+    # According to https://www.geopackage.org/spec/#r24
+    # The column_name column value in a gpkg_geometry_columns row
+    # SHALL be the name of a column in the table or view specified
+    # by the table_name column value for that row.
+    """
+    JOIN gpkg_contents c ON ( g.table_name = c.table_name )
+    """ *
+    # According to https://www.geopackage.org/spec/#r23
+    # gpkg_geometry_columns table_name column SHALL reference values
+    # in the gpkg_contents table_name column for rows with a data_type
+    # of 'features'
+    """
+    WHERE c.data_type = 'features'
+    """ *
+    # According to https://www.geopackage.org/spec/#r146
+    # The srs_id value in a gpkg_geometry_columns table row SHALL
+    # match the srs_id column value from the corresponding row in
+    # the gpkg_contents table.
+    """
+    AND g.srs_id = c.srs_id
+    LIMIT 1 OFFSET ($layer-1);
+    """
+  )
+
+  isnothing(layerinfo) && throw(ErrorException("layer $layer not found in GeoPackage"))
+  metadata = first(layerinfo)
+
+  # According to https://www.geopackage.org/spec/#r33
+  # Feature table geometry columns SHALL contain geometries
+  # with the srs_id specified for the column by the
+  # gpkg_geometry_columns table srs_id column value.
+  org = uppercase(metadata.org)
+  code = metadata.code
+  srsid = metadata.srsid
+
+  # According to https://www.geopackage.org/spec/#r27
+  # The z value in a gpkg_geometry_columns table row SHALL be one of 0, 1, or 2
+  # 0: z values prohibited; 1: z values mandatory; 2: z values optional
+  z = isone(metadata.z)
+  crs = if srsid == 0 || srsid == 99999
+    # An srs_id of 0 SHALL be used for undefined geographic coordinate reference systems
+    # An srs_id of 99999 is recognized as a placeholder code, we will default to undefined geographic crs
+    z ? LatLonAlt{WGS84Latest} : LatLon{WGS84Latest}
+  elseif srsid == -1
+    # An srs_id of -1 SHALL be used for undefined Cartesian coordinate reference systems
+    z ? Cartesian3D{NoDatum} : Cartesian{NoDatum}
+  else
+   # org is a case-insensitive name of the defining organization e.g. EPSG or epsg
+   # code is a numeric ID of the spatial reference system assigned by the organization
+    if code in (0,-1)
```

Comment:
I mean these lines 98 to 101 could be a single `elseif`? Also, the checks with `org == "EPSG"` seem to ignore the case of the string. Should we use `uppercase(org)` in the comparisons?